tardis/montecarlo/src/integrator.c

```diff
@@ -258,11 +270,23 @@ _formal_integral(
                        ++patt_S_ul)
                     {
                       if (*pline < nu_end)
+                        // Calculate e-scattering optical depth to grid cell boundary
+                        zend = storage->time_explosion / C_INV * (1. - nu_end / nu);
+                        dtau += (zend - zstart) * escat_op;
                         break;
```

Comment:
This break get's called *every* loop because the if statement is not followed by a scope `{}`